### PR TITLE
Profit from open6 client's advanced configuration for connectivity check

### PR DESCRIPTION
**Pull request by parasxos**

### CODE REVIEW

**Review by pnikiel — COMMENTED** [review]
Very elegant implementation IMO.
I have one general question: UA_Client_run_iterate returns the status code. What if something happens to the connection (for example, server stopped responding) w/o calling disconnect? Won't run_iterate exit instantly thus creating 100% CPU-greedy loop? Can we do some basics tests on it (e.g. using iptables to break the connection?), and last but not least, can we at least log errors returned from run_iterate and if there are errors slow down the loop with some usleep?

Is the usage of std::async and std::future well within CC7 baseline compiler? If not, I will not object but we will need to make a clear transition to post-CC7 era.

File: src/uaclient/uasession.cpp [hunk 1]
```diff
@@ -55,9 +57,17 @@ UaSession::~UaSession ()
...
-    while(m_clientRunIterateToggle) {
-        UA_Client_run_iterate(m_client, 1000);
+    
+    while(m_clientRunIterateToggle) 
+    {
+        std::this_thread::sleep_for(std::chrono::milliseconds(500));
+        UaStatus status = UA_Client_run_iterate(m_client, 1000);
+        if( !status.isGood() )
+        {
+            LOG(Log::ERR) << "in UaSession::clientRunIterateThread() " << status.toString().toUtf8();
```
Comment (pnikiel) [inline src/uaclient/uasession.cpp:67]: I guess the log component is missing ?

Comment (parasxos) [reply src/uaclient/uasession.cpp:67]: Not, the UaSession class logs without a specific log component.

Comment (pnikiel) [reply src/uaclient/uasession.cpp:67]: I see, we just did not unify it:
https://github.com/quasar-team/open62541-compat/blob/master/src/logit_logger.cpp#L19
okay, so merge it please and we can unity some other days ;-)

**Pull request merged**